### PR TITLE
Margin tests

### PR DESCRIPTION
**Pull request by bachstatter**
Added some TODO where I think it's worth having a chat.. 
Also found out about potential deposit/ withdraw fees :(

### CODE REVIEW

File: contracts/modules/MarginModule.sol [hunk 1]
```diff
@@ -54,12 +54,13 @@ contract MarginModule is IMarginModule {
         PerpMarketConfiguration.GlobalData storage globalConfig
     ) private {
         IERC20(collateralType).transferFrom(msg.sender, address(this), amount);
+        uint256 feeAmount;
         if (collateralType == address(globalConfig.usdToken)) {
-            globalConfig.synthetix.depositMarketUsd(marketId, address(this), amount);
+            feeAmount = globalConfig.synthetix.depositMarketUsd(marketId, address(this), amount);
```
Comment (davidvuong) [inline contracts/modules/MarginModule.sol:59]: 🤔 This is interesting... I'm not sure about this bit. I assume `feeRatio` is defined on each pool? Why would we have a pool that charges on deposits for this market? If we do charge them (I don't see why we would tho), shouldn't we also update internal account to `margin - feeAmount`. Finally, the core system already emits a `MarketSystemFeePaid` event, do we need to include that again here?

File: contracts/modules/MarginModule.sol [hunk 2]
```diff
@@ -276,6 +277,17 @@ contract MarginModule is IMarginModule {
      * @inheritdoc IMarginModule
      */
     function getCollateralUsd(uint128 accountId, uint128 marketId) external view returns (uint256) {
+        Account.exists(accountId);
```
Comment (davidvuong) [inline contracts/modules/MarginModule.sol:280]: 🙏 Good catch

File: contracts/modules/MarginModule.sol (hunk 1 above)
Comment (bachstatter) [reply contracts/modules/MarginModule.sol:59]: > Edit: Oh is this a global config managed by the core system? Hmm let's talk about this one

yea worth checking that perps gets this accounting done correctly too

Comment (bachstatter) [reply contracts/modules/MarginModule.sol]: Had a chat with sunny. This amount is minted. So it doesn't affect the market's accounting/ traders. If turned on its a fee from LPs to the core system.
So removing my feeAmount changes

File: test/integration/modules/MarginModule.test.ts [hunk 3]
```diff
@@ -668,24 +675,172 @@ describe('MarginModule', async () => {
...
+        .sub(order.keeperFee)
+        .add(order.keeperFeeBufferUsd)
+        .add(newPnl)
+        .add(accruedFunding);
+      // Assert marginUSD after price update
+      assertBn.equal(marginUsdAfterPriceChange, expectedMarginUsdAfterPriceChange.toBN());
+    });
+
+    it('should return 0 for underwater position not yet flagged', async () => {
+      const { PerpMarketProxy } = systems();
```
Comment (bachstatter) [inline test/integration/modules/MarginModule.test.ts:738]: @davidvuong 
Note here that this differ from perps v3. `getAvailableMargin` in perps v3 will return a negative number if the pnl has taken you underwater

**Pull request merged**